cpp/src/gandiva/annotator.cc

```diff
@@ -46,15 +46,23 @@ FieldDescriptorPtr Annotator::MakeDesc(FieldPtr field, bool is_output) {
   int data_idx = buffer_count_++;
   int validity_idx = buffer_count_++;
   int offsets_idx = FieldDescriptor::kInvalidIdx;
+  int child_offsets_idx = FieldDescriptor::kInvalidIdx;
```

Comment:
as for list type array data:
- list<binary-like> array data structure:
![image](https://user-images.githubusercontent.com/76817639/103439124-4f48fc80-4c75-11eb-89d5-9b94987e84f3.png)
- list<primitive> array data structure:
![image](https://user-images.githubusercontent.com/76817639/103439220-06457800-4c76-11eb-99ab-7a55751e6941.png)